cmd/abapEnvironmentRunATCCheck.go

```diff
@@ -244,19 +244,30 @@ func logAndPersistATCResult(body []byte, atcResultFileName string, generateHTML
 
 func runATC(requestType string, details abaputils.ConnectionDetailsHTTP, body []byte, client piperhttp.Sender) (*http.Response, error) {
 
-	log.Entry().WithField("ABAP endpoint: ", details.URL).Info("Triggering ATC run")
+	log.Entry().WithField("ABAP endpoint: ", details.URL).Info("triggering ATC run")
 
 	header := make(map[string][]string)
 	header["X-Csrf-Token"] = []string{details.XCsrfToken}
 	header["Content-Type"] = []string{"application/vnd.sap.atc.run.parameters.v1+xml; charset=utf-8;"}
 
-	req, err := client.SendRequest(requestType, details.URL, bytes.NewBuffer(body), header, nil)
-	if err != nil {
+	resp, err := client.SendRequest(requestType, details.URL, bytes.NewBuffer(body), header, nil)
+	logResponseBody(resp)
+	if err != nil || resp.StatusCode == 400 { //send request does not seem to produce error with StatusCode 400!!!
+		err = abaputils.HandleHTTPError(resp, err, "triggering ATC run failed", details)
 		log.SetErrorCategory(log.ErrorService)
-		return req, fmt.Errorf("Triggering ATC run failed: %w", err)
+		return resp, fmt.Errorf("triggering ATC run failed: %w", err)
 	}
-	defer req.Body.Close()
-	return req, err
+	defer resp.Body.Close()
+	return resp, err
+}
+
+func logResponseBody(resp *http.Response) error {
```

Comment:
As far as I know, the response of an http request can be nil (if the ABAP does not send a response).